session.go

```diff
@@ -345,9 +336,6 @@ func (s *session) Retry() error {
 // Unlike normal Exec, it doesn't reset statement status, doesn't commit or rollback the current transaction
 // and doesn't write binlog.
 func (s *session) ExecRestrictedSQL(ctx context.Context, sql string) (ast.RecordSet, error) {
-	if err := s.checkSchemaValidOrRollback(); err != nil {
```

Comment:
It's ok, `sql1(it's fail to execute);` will cause transaction rollback, then sql2 execute in a new transaction and success.